PINCache/PINDiskCache.m

```diff
@@ -113,12 +113,14 @@ - (instancetype)initWithName:(NSString *)name rootPath:(NSString *)rootPath
         _cacheURL = [NSURL fileURLWithPathComponents:@[ rootPath, pathComponent ]];
         
         //we don't want to do anything without setting up the disk cache, but we also don't want to block init, it can take a while to initialize
+        __weak typeof(self) weakSelf = self;
```

Comment:
I'm sure it's not retaining, but the purpose of the change is silencing the warning

Reply:
What is the warning specifically? Is it about accessing the ivar without referencing self? If so the fix is to use `self->_instanceLock` instead of `_instanceLock`.

Reply:
Here is an image of the exact warnings I get if I use `self->` or if I leave it as is

<img width="666" alt="screen shot 2018-03-22 at 4 38 57 pm" src="https://user-images.githubusercontent.com/13370621/37797291-c7521998-2def-11e8-9ca7-63138cf62785.png">
<img width="863" alt="screen shot 2018-03-22 at 4 39 11 pm" src="https://user-images.githubusercontent.com/13370621/37797293-c76847c2-2def-11e8-9171-ac14f6225170.png">

Reply:
Using `self->` fixes the warning as well.
new commit removes the `strongSelf` and uses `self` instead

Reply:
Do you mind getting rid of all the weakSelfs as well? I don't think they're necessary to fix that warning.